lib/matplotlib/backend_tools.py

```diff
@@ -335,6 +335,16 @@ def send_message(self, event):
             except (ValueError, OverflowError):
                 pass
             else:
+                artists = [a for a in event.inaxes.mouseover_set
+                           if a.contains(event) and a.get_visible()]
+
+                if artists:
+                    a = max(enumerate(artists), key=lambda x: x[1].zorder)[1]
```

Comment:
I looked at this earlier today and have a memory of doing it this way intentionally, but no longer remember the _why_.

The discussion at https://github.com/matplotlib/matplotlib/pull/4813/files is not enlightening and I can not even claim I just carried through some older code.  

I think making that change is fine.